cmd/nvidia-dra-controller/imex.go

```diff
@@ -334,7 +353,7 @@ func (offsets imexDomainOffsets) add(imexDomainID string, cliqueID string, resou
 
 	// If we reach the limit, return an error
 	if offset == driverImexChannelLimit {
-		return -1, fmt.Errorf("channel limit reached")
+		return -1, transientError{fmt.Errorf("channel limit reached")}
```

Comment:
As per my last comment, we also need to wrap the error of  func `splitImexDomain` in case the label value string is malformed

Reply:
Why would we wrap that as transient error though?  It's not a transient error, its a real error, so we don't want to retry when it occurs.